datahub-web-react/src/app/entity/view/builder/utils.ts

```diff
@@ -19,6 +25,35 @@ export const extractEntityTypesFilterValues = (filters: Array<FacetFilterInput>)
         .flatMap((filter) => filter.values as EntityType[]);
 };
 
+/**
+ * Converts the nested subtype filter to be split into entity type filters and subType filters.
+ * Right now we don't allow mixing of entity type and subType filters when creating a view since
+ * this filter requires an OR between entity type and subType but AND between other filter types (like tags).
+ */
+export function convertNestedSubTypeFilter(filters: Array<FacetFilterInput>) {
+    const convertedFilters = filters.filter((f) => f.field !== ENTITY_SUB_TYPE_FILTER_NAME) || [];
+    const nestedSubTypeFilter = filters.find((f) => f.field === ENTITY_SUB_TYPE_FILTER_NAME);
```

Comment:
do we need 2 variables with the same value here? 

let's do a lot of manual testing to make sure the views feature is still working as expected

Reply:
two diferent values actually - one filters where the field is **not** `ENTITY_SUB_TYPE_FILTER_NAME` and the other is finding the filter that is `ENTITY_SUB_TYPE_FILTER_NAME`.

oh yes sir i've been manually testing this guy pretty hard.

I explained outside of this PR, but for right now if you try to create a view from filters, if you mix nested subtypes and top level entity types, you get an error message in the UI since we don't support that type of OR/AND logic right now (tag = tag1 AND ((entity = Dataset AND typeNames = Table) OR entity = container))

but if you just have entity types or just subtypes, we allow you to create a view and that's where this function comes in and converts those to the appropriate filters for a view